cmd/abapEnvironmentRunATCCheck.go

```diff
@@ -91,91 +89,113 @@ func handleATCresults(resp *http.Response, details abaputils.ConnectionDetailsHT
-	if err == nil {
-		checkVariantString, packageString, softwareComponentString, err = buildATCCheckBody(ATCConfig)
-	}
+func triggerATCRun(config abapEnvironmentRunATCCheckOptions, details abaputils.ConnectionDetailsHTTP, client piperhttp.Sender) (*http.Response, error) {
 
-	//Trigger ATC run
+	bodyString, err := getATCRequestBody(config)
+	if err != nil {
+		return nil, err
+	}
 	var resp *http.Response
-	var bodyString = `<?xml version="1.0" encoding="UTF-8"?><atc:runparameters xmlns:atc="http://www.sap.com/adt/atc" xmlns:obj="http://www.sap.com/adt/objectset"` + checkVariantString + `><obj:objectSet>` + softwareComponentString + packageString + `</obj:objectSet></atc:runparameters>`
+	abapEndpoint := details.URL
+
 	log.Entry().Debugf("Request Body: %s", bodyString)
 	var body = []byte(bodyString)
-	if err == nil {
-		details.URL = abapEndpoint + "/sap/bc/adt/api/atc/runs?clientWait=false"
-		resp, err = runATC("POST", details, body, client)
-	}
+	details.URL = abapEndpoint + "/sap/bc/adt/api/atc/runs?clientWait=false"
+	resp, err = runATC("POST", details, body, client)
+	return resp, err
+}
+
+func getATCRequestBody(config abapEnvironmentRunATCCheckOptions) (bodyString string, err error) {
+
+	atcConfig, err := resolveATCConfiguration(config)
 	if err != nil {
-		return resp, fmt.Errorf("Triggering ATC run failed: %w", err)
+		return "", err
+	}
+
+	// Create string for the run parameters
+	variant := "ABAP_CLOUD_DEVELOPMENT_DEFAULT"
+	if atcConfig.CheckVariant != "" {
+		variant = atcConfig.CheckVariant
+	}
+	log.Entry().Infof("ATC Check Variant: %s", variant)
+	runParameters := ` checkVariant="` + variant + `"`
+	if atcConfig.Configuration != "" {
+		runParameters += ` configuration="` + atcConfig.Configuration + `"`
+	}
+
+	objectSet, err := getATCObjectSet(atcConfig)
+
+	bodyString = `<?xml version="1.0" encoding="UTF-8"?><atc:runparameters xmlns:atc="http://www.sap.com/adt/atc" xmlns:obj="http://www.sap.com/adt/objectset"` + runParameters + `>` + objectSet + `</atc:runparameters>`
+	return bodyString, err
+}
+
+func resolveATCConfiguration(config abapEnvironmentRunATCCheckOptions) (atcConfig ATCConfiguration, err error) {
+
+	if config.AtcConfig != "" {
+		// Configuration defaults to AUnitConfig
+		atcConfigFile, err := abaputils.ReadConfigFile(config.AtcConfig)
+		if err != nil {
+			return atcConfig, err
+		}
+		json.Unmarshal(atcConfigFile, &atcConfig)
+		return atcConfig, err
+
+	} else if config.Repositories != "" {
+		// Fallback / EasyMode is the Repositories configuration
+		repos, err := abaputils.GetRepositories((&abaputils.RepositoriesConfig{Repositories: config.Repositories}))
+		if err != nil {
+			return atcConfig, err
+		}
+		for _, repo := range repos {
+			atcConfig.Objects.SoftwareComponent = append(atcConfig.Objects.SoftwareComponent, SoftwareComponent{Name: repo.Name})
+		}
+		return atcConfig, nil
+	} else {
+		// Fail if no configuration is provided
+		return atcConfig, errors.New("No configuration provided")
```

Comment:
Maybe we should give the user more hints on which configuration was missing, e.g. `atcConfig or Repositories are missing. Please specify either of these parameters`